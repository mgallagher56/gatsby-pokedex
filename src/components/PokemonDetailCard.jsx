import React from "react"
import { GatsbyImage } from 'gatsby-plugin-image'
import { makeStyles } from '@material-ui/core/styles'

import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import { CardMedia } from '@material-ui/core'

// styles
const useStyles = makeStyles((theme) => ({
  card: {
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    boxShadow: '0px 0px 10px rgba(0, 0, 0, 0.2)'
  },
  cardTitle: {
    textTransform: 'capitalize'
  },
  cardContent: {
    flexGrow: 1,
    display: 'flex',
    alignItems: 'left',
    flexDirection: 'column'
  }
}))

/**
 * PokemonDetailCard renders a card component to display pokemon info
 *
 * Props:
 *
 * title - adds title to card
 *
 * description - add a paragraph to provide further detail
 *
 * gatsbyImg - adds gatbsy image object for responsive images
 *
 * fallbackImg - adds a fallback image to render if gatsby image object fails
 *
 * id - adds id of pokemon
 */
const PokemonDetailCard = (props) => {
  // destructure styles for increased legibility
  const {
    card,
    cardTitle,
    cardMedia,
    cardContent
  } = useStyles()

  // markup
  return (
    <Card className={`${card} pokemon-detail-card`}>
      {props.gatsbyImg
        ? <GatsbyImage
            objectFit='contain'
            className={cardMedia}
            image={props.gatsbyImg}
            title={props.title}
            alt={props.title}
          />
        : <CardMedia
            className={cardMedia}
            image={props.fallbackImg}
            title={props.title}
            alt={props.title}
          />}
      <CardContent className={cardContent}>
        <Typography variant='h5' component='h2' className={cardTitle}>
          {props.title}
        </Typography>
        <Typography variant='body1' component='p'>
          {props.description}
        </Typography>
        <Typography variant='body1' component='p'>
          Pokemon No. {props.id}
        </Typography>
      </CardContent>
    </Card>
  )
}

export default PokemonDetailCard
