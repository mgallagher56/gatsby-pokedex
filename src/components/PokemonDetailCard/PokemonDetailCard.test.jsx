import React from 'react'
import { render, cleanup } from '@testing-library/react'
import PokemonDetailCard from './PokemonDetailCard'

describe('PokemonDetailCard Component', () => {
  afterEach(cleanup)

  it('renders PokemonDetailCard Component', () => {
    render(<PokemonDetailCard />)
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<PokemonDetailCard />)
    expect(asFragment(<PokemonDetailCard />)).toMatchSnapshot()
  })

  it('should have Hello world title', () => {
    const { getByTestId } = render(<PokemonDetailCard title='Hello world' />)
    expect(getByTestId('pokemon-detail-card-title')).toMatchInlineSnapshot(`
    <h2
      class="MuiTypography-root makeStyles-cardTitle-8 MuiTypography-h5"
      data-testid="pokemon-detail-card-title"
    >
      Hello world
    </h2>
    `)
  })

  it('should have a title', () => {
    const { queryByTestId } = render(<PokemonDetailCard title='title text' />)
    expect(queryByTestId('pokemon-detail-card-title')).toBeTruthy()
  })

  it('should not have a title', () => {
    const { queryByTestId } = render(<PokemonDetailCard />)
    expect(queryByTestId('pokemon-detail-card-title')).toBeNull()
  })

  it('should have a description', () => {
    const { queryByTestId } = render(<PokemonDetailCard description='lorem ipsum' />)
    expect(queryByTestId('pokemon-detail-card-description')).toBeTruthy()
  })

  it('should not have a description', () => {
    const { queryByTestId } = render(<PokemonDetailCard />)
    expect(queryByTestId('pokemon-detail-card-description')).toBeNull()
  })
})
