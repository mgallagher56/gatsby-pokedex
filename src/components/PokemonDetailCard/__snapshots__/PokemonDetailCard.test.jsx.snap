// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetailCard Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-4 pokemon-detail-card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <img
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
    />
    <div
      class="MuiCardContent-root makeStyles-cardContent-6"
    />
  </div>
</DocumentFragment>
`;
